src/components/Editor/Runners/PythonRunner/PythonRunner.jsx

```diff
@@ -412,31 +431,37 @@ const PythonRunner = () => {
                 </TabPanel>
               </Tabs>
             </div>
-          ) : null}
-          <div className="output-panel output-panel--text">
-            <Tabs forceRenderTabPanel={true}>
-              <div className="react-tabs__tab-container">
-                <TabList>
-                  <Tab key={0}>
-                    <span className="react-tabs__tab-text">
-                      {t("output.textOutput")}
-                    </span>
-                  </Tab>
-                </TabList>
-                {!hasVisualOutput && !isEmbedded && isMobile ? (
-                  <RunnerControls skinny />
-                ) : null}
-              </div>
-              <ErrorMessage />
-              <TabPanel key={0}>
-                <pre
-                  className={`pythonrunner-console pythonrunner-console--${settings.fontSize}`}
-                  onClick={shiftFocusToInput}
-                  ref={output}
-                ></pre>
-              </TabPanel>
-            </Tabs>
-          </div>
+          )}
+          {showTextOutput && (
+            <div className={outputPanelClasses("text")}>
+              <Tabs forceRenderTabPanel={true}>
+                <div
+                  className={classNames("react-tabs__tab-container", {
+                    "react-tabs__tab-container--hidden": singleOutputPanel,
+                  })}
+                >
+                  <TabList>
+                    <Tab key={0}>
+                      <span className="react-tabs__tab-text">
+                        {t("output.textOutput")}
+                      </span>
+                    </Tab>
+                  </TabList>
+                  {!hasVisualOutput && !isEmbedded && isMobile ? (
+                    <RunnerControls skinny />
+                  ) : null}
```

Comment:
⛏️ I'm sure I've tried this before but it might not work... can we get rid of this null?
```suggestion
                  {!hasVisualOutput && !isEmbedded && isMobile && (
                    <RunnerControls skinny />
                  )}
```